Mathlib/Algebra/Homology/ShortComplex/RightHomology.lean

```diff
@@ -891,6 +891,233 @@ noncomputable def fromOpcyclesNatTrans :
+induces a right homology map data in the original category. -/
+@[simps]
+def LeftHomologyMapData.unop {S₁ S₂ : ShortComplex Cᵒᵖ} {φ : S₁ ⟶ S₂}
+    {h₁ : S₁.LeftHomologyData} {h₂ : S₂.LeftHomologyData}
+    (ψ : LeftHomologyMapData φ h₁ h₂) : RightHomologyMapData (unopMap φ) h₂.unop h₁.unop where
+  φQ := ψ.φK.unop
+  φH := ψ.φH.unop
+  commp := Quiver.Hom.op_inj (by simp)
+  commg' := Quiver.Hom.op_inj (by simp)
+  commι := Quiver.Hom.op_inj (by simp)
+
+/-- A right homology map data for a morphism of short complexes induces
+a left homology map data in the opposite category. -/
+@[simps]
+def RightHomologyMapData.op {S₁ S₂ : ShortComplex C} {φ : S₁ ⟶ S₂}
+    {h₁ : S₁.RightHomologyData} {h₂ : S₂.RightHomologyData}
+    (ψ : RightHomologyMapData φ h₁ h₂) : LeftHomologyMapData (opMap φ) h₂.op h₁.op where
+  φK := ψ.φQ.op
+  φH := ψ.φH.op
+  commi := Quiver.Hom.unop_inj (by simp)
+  commf' := Quiver.Hom.unop_inj (by simp)
+  commπ := Quiver.Hom.unop_inj (by simp)
+
+/-- A right homology map data for a morphism of short complexes in the opposite category
+induces a left homology map data in the original category. -/
+@[simps]
+def RightHomologyMapData.unop {S₁ S₂ : ShortComplex Cᵒᵖ} {φ : S₁ ⟶ S₂}
+    {h₁ : S₁.RightHomologyData} {h₂ : S₂.RightHomologyData}
+    (ψ : RightHomologyMapData φ h₁ h₂) : LeftHomologyMapData (unopMap φ) h₂.unop h₁.unop where
+  φK := ψ.φQ.unop
+  φH := ψ.φH.unop
+  commi := Quiver.Hom.op_inj (by simp)
+  commf' := Quiver.Hom.op_inj (by simp)
+  commπ := Quiver.Hom.op_inj (by simp)
+
+variable (S)
+
+/-- The right homology in the opposite category of the opposite of a short complex identifies
+to the left homology of this short complex. -/
+noncomputable def rightHomologyOpIso [S.HasLeftHomology] :
+    S.op.rightHomology ≅ Opposite.op S.leftHomology :=
+  S.leftHomologyData.op.rightHomologyIso
+
+/-- The left homology in the opposite category of the opposite of a short complex identifies
+to the right homology of this short complex. -/
+noncomputable def leftHomologyOpIso [S.HasRightHomology] :
+    S.op.leftHomology ≅ Opposite.op S.rightHomology :=
+  S.rightHomologyData.op.leftHomologyIso
+
+/-- The opcycles in the opposite category of the opposite of a short complex identifies
+to the cycles of this short complex. -/
+noncomputable def opcyclesOpIso [S.HasLeftHomology] :
+    S.op.opcycles ≅ Opposite.op S.cycles :=
+  S.leftHomologyData.op.opcyclesIso
+
+/-- The cycles in the opposite category of the opposite of a short complex identifies
+to the opcycles of this short complex. -/
+noncomputable def cyclesOpIso [S.HasRightHomology] :
+    S.op.cycles ≅ Opposite.op S.opcycles :=
+  S.rightHomologyData.op.cyclesIso
+
+@[simp]
+lemma leftHomologyMap'_op
+    (φ : S₁ ⟶ S₂) (h₁ : S₁.LeftHomologyData) (h₂ : S₂.LeftHomologyData) :
+    (leftHomologyMap' φ h₁ h₂).op = rightHomologyMap' (opMap φ) h₂.op h₁.op := by
+  let γ : LeftHomologyMapData φ h₁ h₂ := default
```

Comment:
I'm not a huge fan of the `default` here, especially since you have defined a name for this.

```suggestion
  let γ : LeftHomologyMapData φ h₁ h₂ := leftHomologyMapData φ h₁ h₂
```